File: frontend/src/types/foundations.ts
export interface Material {
    id?: number;
    sku: string;
    name: string;
    category: string;
    
    // ENUM estricto para coincidir con backend y evitar errores de dedo
    production_route: 'MATERIAL' | 'PROCESO' | 'CONSUMIBLE' | 'SERVICIO';
    
    purchase_unit: string;
    usage_unit: string;
    conversion_factor: number;
    current_cost: number;
    
    // Vinculación
    associated_element_sku?: string | null;
    provider_id?: number | null;

    // Stocks (Backend siempre los envía, aunque sean 0)
    physical_stock: number;
    committed_stock: number; 

    // Control
    is_active: boolean;
}

export interface Provider {
    id?: number;
    
    // Datos Generales
    business_name: string;
    legal_name?: string;
    rfc_tax_id?: string;
    email?: string;
    phone?: string;
    phone2?: string;           // <--- NUEVO
    
    // Datos del Contacto Principal
    contact_name?: string;
    contact_email?: string;    // <--- NUEVO
    contact_cellphone?: string;// <--- NUEVO
    
    // Comercial
    credit_days: number;
    is_active: boolean;
}

export interface Client {
    id?: number;
    full_name: string;
    email: string;
    phone: string;
    rfc_tax_id?: string;
    fiscal_address?: string;
    
    // Contacto 1
    contact_name?: string;
    contact_phone?: string;
    contact_dept?: string;     // <--- NUEVO
    contact_email?: string;    // <--- NUEVO
    
    // Contacto 2
    contact2_name?: string;    // <--- NUEVO
    contact2_phone?: string;   // <--- NUEVO
    contact2_dept?: string;    // <--- NUEVO
    contact2_email?: string;   // <--- NUEVO
    
    // Contacto 3
    contact3_name?: string;    // <--- NUEVO
    contact3_phone?: string;   // <--- NUEVO
    contact3_dept?: string;    // <--- NUEVO
    contact3_email?: string;   // <--- NUEVO
    
    // Contacto 4
    contact4_name?: string;    // <--- NUEVO
    contact4_phone?: string;   // <--- NUEVO
    contact4_dept?: string;    // <--- NUEVO
    contact4_email?: string;   // <--- NUEVO
    
    notes?: string;
    registration_date?: string;
    is_active: boolean;
}

export interface TaxRate {
    id: number; // Siempre existe al leer configuración
    name: string;
    rate: number;
    is_active: boolean;
}

export interface GlobalConfig {
    id?: number;
    
    // --- VARIABLES DE NEGOCIO ---
    target_profit_margin: number;
    cost_tolerance_percent: number;
    quote_validity_days: number;
    default_edgebanding_factor: number;
    default_tax_rate_id?: number | null;

    // --- METAS FINANCIERAS (Dashboard Directivo) ---
    annual_sales_target: number;
    last_year_sales: number;

    // --- NÓMINA A DESTAJO (Tabulador Global de Instaladores) ---
    default_leader_daily_rate: number;
    default_helper_daily_rate: number;

    // --- IDENTIDAD CORPORATIVA ---
    company_name: string;
    company_rfc?: string;
    company_address?: string;
    company_phone?: string;
    company_email?: string;
    company_website?: string;
    
    // Logo
    logo_path?: string | null; 
    updated_at?: string;
}

File: backend/app/models/foundations.py
from typing import Optional, List, TYPE_CHECKING
from sqlmodel import Field, SQLModel, Relationship
from datetime import datetime

# Evitamos import circular solo para tipado
if TYPE_CHECKING:
    from .material import Material

# --- MODELOS ---

class TaxRate(SQLModel, table=True):
    __tablename__ = "tax_rates"
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    rate: float
    is_active: bool = Field(default=True)

class GlobalConfig(SQLModel, table=True):
    __tablename__ = "global_config"
    id: Optional[int] = Field(default=None, primary_key=True)
    
    # Identidad
    company_name: str
    company_rfc: Optional[str] = None
    company_address: Optional[str] = None
    company_phone: Optional[str] = None
    company_email: Optional[str] = None
    company_website: Optional[str] = None
    logo_path: Optional[str] = None 
    
    # Reglas de Negocio
    target_profit_margin: float
    cost_tolerance_percent: float
    quote_validity_days: int
    default_edgebanding_factor: float

    # Metas Financieras
    annual_sales_target: float = Field(default=0.0)
    last_year_sales: float = Field(default=0.0)
    
    # Relaciones
    default_tax_rate_id: Optional[int] = Field(default=None, foreign_key="tax_rates.id")
    updated_at: datetime = Field(default_factory=datetime.now)

class Provider(SQLModel, table=True):
    __tablename__ = "providers"
    id: Optional[int] = Field(default=None, primary_key=True)
    business_name: str = Field(index=True)
    legal_name: Optional[str] = None
    rfc_tax_id: Optional[str] = None
    contact_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    credit_days: int
    is_active: bool = Field(default=True)
    materials: List["Material"] = Relationship(back_populates="provider")

class Client(SQLModel, table=True):
    __tablename__ = "clients_v2"
    id: Optional[int] = Field(default=None, primary_key=True)
    full_name: str = Field(index=True)
    rfc_tax_id: Optional[str] = None
    email: str
    phone: str
    fiscal_address: Optional[str] = None
    
    # --- CONTACTO PRINCIPAL ---
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    contact_dept: Optional[str] = None  # <--- NUEVO: Departamento

    # --- CONTACTO 2 ---
    contact2_name: Optional[str] = None
    contact2_phone: Optional[str] = None
    contact2_dept: Optional[str] = None

    # --- CONTACTO 3 ---
    contact3_name: Optional[str] = None
    contact3_phone: Optional[str] = None
    contact3_dept: Optional[str] = None

    # --- CONTACTO 4 ---
    contact4_name: Optional[str] = None
    contact4_phone: Optional[str] = None
    contact4_dept: Optional[str] = None

    notes: Optional[str] = None
    registration_date: datetime = Field(default_factory=datetime.now)
    is_active: bool = Field(default=True)